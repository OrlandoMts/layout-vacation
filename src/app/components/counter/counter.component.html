<div class="container">
  <h1>counter {{ counter }}</h1>
  <button class="btn btn-primary" (click)="increaseButton()">
    Incrementar
  </button>
  <button class="btn btn-primary" (click)="decreaseButton()">
    Decrementar
  </button>
</div>
