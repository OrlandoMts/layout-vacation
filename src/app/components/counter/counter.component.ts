import { Component } from '@angular/core';

@Component({
  selector: 'app-counter',
  templateUrl: './counter.component.html',
  styles: [],
})
export class CounterComponent {
  public counter: number = 0;

  constructor() {}

  increaseBy(value: number) {
    this.counter += value;
  }

  increaseButton() {
    this.counter++;
  }

  decreaseButton() {
    this.counter--;
  }
}
