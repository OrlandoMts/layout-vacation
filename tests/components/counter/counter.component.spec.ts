import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from '../../../src/app/components/counter/counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [CounterComponent],
    });
    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should do match to snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  /* it('should increase', () => {
    component.increaseButton
  }); */

  it('should increase and decrease by button', () => {
    const buttons = compiled.querySelectorAll('button');
    buttons[0].click();
    expect(component.counter).toBe(1);
    buttons[1].click();
    expect(component.counter).toBe(0);
  });

  it('should change html-tag h1', () => {
    component.counter += 1;
    fixture.detectChanges();

    const h1 = compiled.querySelector('h1');
    expect(h1?.textContent).toContain('1');
  });
});
