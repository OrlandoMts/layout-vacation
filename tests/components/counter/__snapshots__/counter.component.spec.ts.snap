// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent should do match to snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="container"
  >
    <h1>
      counter 0
    </h1>
    <button
      class="btn btn-primary"
    >
       Incrementar 
    </button>
    <button
      class="btn btn-primary"
    >
       Decrementar 
    </button>
  </div>
</div>
`;
